### PR TITLE
Feature/builder tests

### PR DESCRIPTION
**Pull request by t-eliks**

### CODE REVIEW

File: Core/NoOpRunner.Core.Tests/GameStateBuilderTests.cs [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+using Xunit;
+
+namespace NoOpRunner.Core.Tests
+{
+    public class GameStateBuilderTests
+    {
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void GameStateBuilder_WhenMapIsUninitialized_AndShapeIsAdded_ShouldThrowException(bool initializeMap)
```
Comment (Kodnot) [inline Core/NoOpRunner.Core.Tests/GameStateBuilderTests.cs]: Wouldn't add the '_' in the middle, but that's personal preference, this also works

File: Core/NoOpRunner.Core.Tests/GameStateBuilderTests.cs [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+            }
+
+            // Act
+            var action = new Action(() => gameStateConfigurator.AddShape(f => f.GetShape(Enums.Shape.Saw, 1, 1)));
+
+            // Assert
+            if (!initializeMap)
+                action.ShouldThrow<Exception>();
+            else
+                action.ShouldNotThrow();
```
Comment (Kodnot) [inline Core/NoOpRunner.Core.Tests/GameStateBuilderTests.cs:35]: If you have a conditional statement in your assertion, that's a big red flag that you should split the test into two. Same for a few other tests. You shouldn't have conditionals in your arrange or assert stages, you pretty much shouldn't have conditionals in a test function at all. The code will be a bit more bloated when you split the tests, but easy code comprehension is what matters in tests, code repetition is fine.

File: Core/NoOpRunner.Core.Tests/GameStateBuilderTests.cs (hunk 1 above)
Comment (t-eliks) [reply Core/NoOpRunner.Core.Tests/GameStateBuilderTests.cs]: Done

File: Core/NoOpRunner.Core/Shapes/EntityShapes/Rocket.cs [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 namespace NoOpRunner.Core.Shapes.EntityShapes
 {
-    class Rocket : EntityShape
+    public class Rocket : EntityShape
```
Comment (Kodnot) [inline Core/NoOpRunner.Core/Shapes/EntityShapes/Rocket.cs]: Don't make the classes public, use `InternalsVisibleTo` in AssemblyInfo.cs if need be instead

**Review by Kodnot — CHANGES_REQUESTED** [review]

File: Core/NoOpRunner.Core/Shapes/EntityShapes/Rocket.cs (hunk 3 above)
Comment (t-eliks) [reply Core/NoOpRunner.Core/Shapes/EntityShapes/Rocket.cs]: These classes aren't supposed to be private. They should be public by default and only private explicitly if intended.

File: Core/NoOpRunner.Core.Tests/GameStateBuilderTests.cs (hunk 2 above)
Comment (t-eliks) [reply Core/NoOpRunner.Core.Tests/GameStateBuilderTests.cs:35]: The reason for this is to make sure the same setup but without initialization clause doesn't cause an exception, but without it does. If these tests are split into separate, eventually the arrangement might become different in those two cases and the one that asserts an exception might start passing for the wrong reasons and will no longer bring any value.

**Review by Kodnot — APPROVED** [review]

**Pull request merged**